src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenChatHandler.java

```diff
@@ -45,30 +41,30 @@ public void notifyDeath(final DamageSource damageSource)
         if (citizen.getCitizenColonyHandler().getColony() != null && citizen.getCitizenData() != null)
         {
             final IJob<?> job = citizen.getCitizenJobHandler().getColonyJob();
-            MutableComponent contentComponent;
+            Component nameComponent;
             if (job != null)
             {
-                contentComponent = Component.translatable(
-                  TranslationConstants.WORKER_DIED,
+                nameComponent = Component.translatable(
+                  TranslationConstants.WORKER_FULL,
                   Component.translatable(job.getJobRegistryEntry().getTranslationKey()),
-                  citizen.getCitizenData().getName(),
-                  Math.round(citizen.getX()),
-                  Math.round(citizen.getY()),
-                  Math.round(citizen.getZ()),
-                  Component.translatable(damageSource.msgId));
+                  citizen.getCitizenData().getName());
             }
             else
             {
-                contentComponent = Component.translatable(
-                  TranslationConstants.COLONIST_DIED,
-                  citizen.getCitizenData().getName(),
-                  Math.round(citizen.getX()),
-                  Math.round(citizen.getY()),
-                  Math.round(citizen.getZ()),
-                  Component.translatable(damageSource.msgId));
+                nameComponent = Component.translatable(
+                  TranslationConstants.COLONIST_FULL,
+                  citizen.getCitizenData().getName());
             }
-
-            MessageUtils.format(contentComponent)
+            String result = citizen.getCombatTracker().getDeathMessage().getString().replace(citizen.getDisplayName().getString(), nameComponent.getString());
+            if (isAlphanumeric(result.charAt(result.length() - 1))) {
```

Comment:
what is this for?

Reply:
This gets the actual death message, e.g. `%s was slain by Zombie` or `%s was blown up by Creeper` or `%s fell from a high place`. The piece of code used here is an adaptation of what player death messages use. `%s` is then replaced by the colonist's name and profession.
The position is then appended; it now uses a separate translation. The `if (isAlphanumeric(...))` check is to ensure that something like `The citizen XY fell from a high place Their death position is X Y Z` (with a missing separator between the sentences) doesn't happen.

Reply:
By the looks, I think it is to strip off any non alphanumeric character at the end, and replace it with an exclamation mark. So maybe trying to strip off any punctuation that is already included as part of the death message?

Reply:
Although, it's probably easier to do this through a regex statement